src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -529,4 +532,21 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
             authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
         }
     }
+
+    private void setUserRolesForPostings(Posting posting, Course postingCourse) {
+        User author = userRepository.findOneWithGroupsAndAuthoritiesById(posting.getAuthor().getId()).get();
+
+        if (author.getAuthorities().stream().anyMatch(authority -> authority.equals(Authority.INSTRUCTOR_AUTHORITY) || authority.equals(Authority.ADMIN_AUTHORITY))
```

Comment:
Are you suggesting that we only check roles based on course groups?

Reply:
Yes. Consider the following case:

User1 and user2 both have the "Instructor" authority since they are both instructors in course1. But user2 is also a student who takes part in course2, another course where user1 is instructor. If user1 answers a post for course1 he should have the instructor image, but if he writes something for course2, he should only be shown as a user.

Reply:
I see the point. Can we say that there are no global instructors in Artemis where the user is supposed to be an instructor in all courses? You can think of this as us logging in to TS1 with our TUM IDs.

Reply:
I think global instructors are admins and the `isAtLeastInstrcutor`-method should cover this case.

Reply:
Makes sense to me. Thanks 😊